File: templates/java-webflux/subsystem/user/component/src/main/java/{{PACKAGE_NAME}}/user/UserController.java
package {{PACKAGE_NAME}}.user;

import {{PACKAGE_NAME}}.post.model.Post;
import {{PACKAGE_NAME}}.user.contract.UserWithPosts;
import {{PACKAGE_NAME}}.user.model.User;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/users")
class UserController {

  private UserService userService;

  UserController(UserService userService) {
    this.userService = userService;
  }

  @GetMapping
  Flux<User> findAllUser() {
    return this.userService.findAll();
  }

  @GetMapping("/{id}")
  Mono<UserWithPosts.Res> findUserById(@PathVariable Integer id) {
    return this.userService.findById(id);
  }

  @GetMapping("/{id}/posts")
  Flux<Post> findAllPostByUser(@PathVariable Integer id) {
    return this.userService.findAllPostByUser(id);
  }

}

File: templates/kotlin-webflux/subsystem/user/component/src/main/kotlin/{{PACKAGE_NAME}}/user/UserController.kt
package {{PACKAGE_NAME}}.user

import {{PACKAGE_NAME}}.post.model.Post
import {{PACKAGE_NAME}}.user.contract.UserWithPosts
import {{PACKAGE_NAME}}.user.model.User
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/users")
internal class UserController(
  private val userService: UserService
) {

  @GetMapping
  fun findAllUser(): Flux<User> =
    this.userService.findAll()

  @GetMapping("/{id}")
  fun findUserById(@PathVariable id: Int): Mono<UserWithPosts.Res> =
    this.userService.findById(id)

  @GetMapping("/{id}/posts")
  fun findAllPostByUser(@PathVariable id: Int): Flux<Post> =
    this.userService.findAllPostByUser(id)

}
